src/main/kotlin/ch/unibas/dmi/dbis/cottontail/model/values/ComplexValue.kt

```diff
@@ -0,0 +1,16 @@
+package ch.unibas.dmi.dbis.cottontail.model.values
+
+import ch.unibas.dmi.dbis.cottontail.model.values.complex.Complex
+
+inline class ComplexValue(override val value: Complex) : Value<Complex> {
```

Comment:
From my standpoint, it makes little sense to have a **ComplexValue** that inherits from **Value<Complex>** and that wraps another class **Complex**. This creates a lot of overhead, especially since the latter is an object type. Instead, I would directly make the implementation of **ComplexValue** behave like a complex number.

All the Value<> implementations are optimised for efficiency, because Cottontail DB processes a lot of them. Therefore, the use of inline classes, which provide us with a "thin" wrapper around primitive values (that are still treated like primitive values by the JVM). I would suggest to follow the same paradigm for Complex numbers:  

- Create the Value<Complex> implementation as a wrapper around a 2d float/double array.
- 1st component is the real part, 2nd component is the imaginary part
- Rest (e.g. operations) as defined by the interface